backend/controllers/articleController.js

```diff
@@ -53,16 +54,20 @@ const create = async (req, res) => {
 
 const deleteById = async (req, res) => {
   const id = req.params.id;
+  const userid = req.user["_id"];
```

Comment:
Could this author checking be bypass by a malicious user by intercepting the request and modifying the `_id` if it's too obvious? Don't know if it's possible just thoughts!

Reply:
*Is there a way to quote another file? I can't find how...*
Here is the full `deleteById` function
```js
const deleteById = async (req, res) => {              
  const id = req.params.id;
  const { username } = req.user;
          
  if (!mongoose.Types.ObjectId.isValid(id)) {
    return res.status(404).json({ error: "No such article" });
  }
  const { author } = await Article.findById(id);
 
  if (!author) {
    res.status(400).json({ error: "No such article" });
  } else if (author !== username) {
    res.status(401).json({ error: "Access denied" });
  } else {
    const article = await Article.findByIdAndDelete(id);
    res.status(200).json(article);
  }
};
```
As you can see, there is a test in this function that prevent an attacker to delete an arbitrary article by checking if `author !== user`
So you can modify the value of `_id` if you want but not the value of `author`, it is stored within the article in the database.
By the way, ids are UUID of type 4, which looks like this `bdeb3114-27c9-4025-a473-6952c22bccf5`

Reply:
> Is there a way to quote another file? I can't find how...

Don't know at all maybe a link is the best solution :confused: 

Ok my bad I confused `id` and `user`
Let me reword it:
`author` comes from the DB and is the author of the article `id` we are trying to delete. 
To check if the user can do it we check that the `user` who send this request is the `author`
But if `user` is simply a username like `ctmbl`, when sending the request I can just change it to `atxr` and fake the backend saying that I'm someone that I'm not.
 
Notice that it **totally** depends on the fact that `user` can be modified when sending the request. If `user` is secured by 
the JWT token I think it's safe! 

PS: I hope that my question is not too newbie and makes sense, if it doesn't at all let me know :smiling_face_with_tear:

Reply:
Your question is totally legit and interesting!
At this line, I define the `username` variable.
```
const { username } = req.user;
```
As you can see, it comes from the req.user object, which is defined in the `middleware`.
A `midleware` is a piece of code that will be executed each time the backend receives a request (and before all the others function).
Take a look to the [requireAuth middleware](https://github.com/iScsc/iscsc.fr/pull/2/files#diff-c1ce72598870a1e51768118202e830831f817e460da3e2f2b5900983c6fddd2f)
These two lines confirm that the user id comes from the JWT, so you cannot manipulate it.
```js
const { _id } = jwt.verify(token, process.env.JWT_SECRET);
req.user = await User.findOne({ _id }).select("_id username");
```

Reply:
Thanks a lot for clarifying this point! I actually needed such an explanation to better understand authentication!